fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/export/S3DatatableReportExportServiceImpl.java

```diff
@@ -51,9 +52,20 @@ public ResponseHolder export(String reportName, MultivaluedMap<String, String> q
                     isSelfServiceUserReport);
             try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
                 output.write(byteArrayOutputStream);
+                byte[] fileBytes = byteArrayOutputStream.toByteArray();
+                if (fileBytes.length == 0) {
+                    throw new IllegalStateException("Can not upload empty report content to S3");
```

Comment:
What this will be resolved? It is still HTTP 500, no? Can you please rather use `GeneralPlatformDomainRuleException`?